python/pyarrow/dataset.py

```diff
@@ -852,6 +852,8 @@ def file_visitor(written_file):
         dataset.  The first time each partition directory is encountered
         the entire directory will be deleted.  This allows you to overwrite
         old partitions completely.
+    create_dir : bool, default True
+        Flag to restrict or allow creating directory while writing a dataset.
```

Comment:
```suggestion
    create_dir : bool, default True
        If False, directories will not be created.  This can be useful for filesystems
        that do not require directories.
```

Reply:
Made the change.